src/main/java/org/folio/circulation/domain/RequestQueue.java

```diff
@@ -86,9 +86,17 @@ public Boolean contains(Request request) {
   Collection<Request> getRequestsWithChangedPosition() {
     return requests.stream()
       .filter(Request::hasChangedPosition)
+      // order by position descending
+      .sorted((req1, req2) -> req2.getPosition().compareTo(req1.getPosition()))
       .collect(Collectors.toList());
   }
 
+  public Boolean positionTaken(Request request) {
```

Comment:
This seems to check if the position was previously taken by another request. Is that a reasonable understanding?
```suggestion
  public Boolean positionPreviouslyTaken(Request request) {
```

Reply:
Yes, that is correct. I like the name suggestion. Will update.